plugins/modules/files/archive.py

```diff
@@ -137,6 +137,16 @@
         The state of the input C(path).
     type: str
     returned: always
+dest_state:
+    description:
+      - The state of the I(dest) file.
+      - C(absent) when the file does not exist.
+      - C(archive) when the file is an archive.
+      - C(compress) when the file is compressed, but not an archive.
```

Comment:
Can you elaborate on this, how `archive` module creates a file which is not an `archive`, but yet it is compressed, what is its extension?
I can understand an archive without compression, but a compressed file which has not archived files, what is that?

Reply:
gzip file => file.gz is not an archive

Reply:
@russoz aha, thanks!

Reply:
Yeah, these states are from the module's original `state` return value which was being overwritten by `AnsibleModule`. I think one reason for this might've been when using this module in conjunction with `ansible.builtin.unarchive` which is only possible when the destination is an archive.